### PR TITLE
[fix] at::view

### PR DESCRIPTION
**Pull request by jerryzh168**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#23452 [fix] at::view**

accidently calls clone, but what we want is creating an empty tensor and set storage.

Differential Revision: [D16442756](https://our.internmc.facebook.com/intern/diff/D16442756/)

### CODE REVIEW

File: aten/src/ATen/native/TensorShape.cpp [hunk 1]
```diff
@@ -898,20 +898,38 @@ Tensor numpy_T(const Tensor &self) {
...
+  auto self_ = self.alias();
+  self_.set_(
+    self.storage(), self.storage_offset(), inferred_size, stride_value);
+  return self_;
+}
+
+Tensor alias(const Tensor& self) {
+  Tensor self_;
+  if (self.is_quantized()) {
+    TORCH_CHECK(self.qscheme() == kPerTensorAffine,
```
Comment (zafartahirov) [inline aten/src/ATen/native/TensorShape.cpp]: Should we allow `kPerTensorSymmetric` in here as well?

**Review by z-a-f — APPROVED** [review]
LGTM

File: aten/src/ATen/native/TensorShape.cpp (hunk 1 above)
Comment (jerryzh168) [reply aten/src/ATen/native/TensorShape.cpp]: we don't support that right now

**Comment by jerryzh168** [comment]
@pytorchb retest this please

File: aten/src/ATen/native/TensorShape.cpp (hunk 1 above)
Comment (gchanan) [reply aten/src/ATen/native/TensorShape.cpp]: nit: shouldn't the QTensorImpl constructor check that then?

File: aten/src/ATen/test/basic.cpp [hunk 2]
```diff
@@ -283,6 +283,14 @@ void TestNegativeDim(DeprecatedTypeProperties& type) {
   ASSERT_ANY_THROW(tensor.reshape({-5, -5}));
 }
 
+void TestView(DeprecatedTypeProperties& type) {
```
Comment (gchanan) [inline aten/src/ATen/test/basic.cpp]: nit: would be good to comment why you need this test; that this tests the *Tensor* view path, not the *Variable* view path, which has different semantics (you could link this PR for more details).

**Review by gchanan — APPROVED** [review]

File: aten/src/ATen/native/TensorShape.cpp (hunk 1 above)
Comment (jerryzh168) [reply aten/src/ATen/native/TensorShape.cpp]: I think we have per channel quantizer right now. looking again, the following code actually works for whatever tensor we have, so we can remove this check.

**Pull request closed without merging**